src/dynamic_boundary_conditions/rainfall_data_from_hirds.py

```diff
@@ -86,24 +86,59 @@ def get_data_from_hirds(site_id: str, idf: bool) -> str:
     return site_data
 
 
-def store_data_to_csv(site_id: str, file_path_to_store, idf: bool):
+def get_layout_structure_of_data(site_data: str) -> List[Tuple[int, float, str]]:
     """
-    Store the rainfall data in the form of CSV file in the desired path.
+    Return a list of tuples (skip_rows, rcp, time_period) of the fetched rainfall data's layout structure.
 
     Parameters
     ----------
+    site_data : str
+        Fetched rainfall data text string from the HIRDS website for the requested rainfall site.
+    """
+    skip_rows = []
+    rcp = []
+    time_period = []
+    # Read the site_data text string line by line with a for loop
+    for index, line in enumerate(StringIO(site_data)):
+        # Get lines that contain "(mm) ::" for depth data or "(mm/hr) ::" for intensity data
+        if (line.find("(mm) ::") != -1) or (line.find("(mm/hr) ::") != -1):
+            # Add the row number to skip_rows list
+            skip_rows.append(index + 1)
+            # Add the obtained rcp and time_period values to list
+            rcp_result = re.search(r"(\d*\.\d*)", line)
+            period_result = re.search(r"(\d{4}-\d{4})", line)
+            if rcp_result or period_result is not None:
```

Comment:
Should this be:
```suggestion
            if rcp_result is not None or period_result is not None:
```
or alternatively
```suggestion
            if any(result is not None for result in [rcp_result, period_result]):
```
Its very unlikely to cause an issue either way but I like to be explicit